### ISSUE TITLE
user-data

### ISSUE THREAD

**Issue by jgroom33** [vyos/vyos-cloud-init#11]
does this package enable a config to be defined in user-data similar to: https://github.com/vyos/vyos-cloudinit/tree/master#configuration-with-vyos-config

**Comment by codello** [comment]
I have been playing around with the VyOS cloud-init modules lately. To me it seems that directly supplying a `#vyos-config` document does not work. At least I haven't been able to get a `#vyos-config` document to work and I think this is the intended behavior:
```shell
vyos@vyos:~$ cloud-init devel make-mime --list-types
cloud-boothook
cloud-config
cloud-config-archive
cloud-config-jsonp
jinja2
part-handler
upstart-job
x-include-once-url
x-include-url
x-shellscript
```

However supplying a configuration file is possible via cloud-config (or probably via custom part-handlers) using the `write_files` module:
```yaml
write_files:
- path: /opt/vyatta/etc/config/config.boot
  content: |
    <config.boot here>
```
Note that the `write_files` module is not enabled in the default `vyos-cloud-init` configuration. In order for this to work it should be run after the `vyos` module (otherwise the `vyos` module would overwrite some of the configuration).


Getting a `vbash` script to work is slightly more complicated. First you have to enable the `scripts_user` module (side note: The default configuration includes the `runcmd` module. However `runcmd` commands are only executed if the `scripts_user` module is enabled). This is actually enough to be able to run scripts. Unfortunately the `cloud-final` stage will be run before the VyOS system is fully initialized which results in strange errors and invalid configs if you try to modify the configuration. The problem seems to be that the `vyos-router` systemd service has to be `Type=simple` which means that the system will be initialized *some time* after the service is started. Fortunately the file `/tmp/vyos-config-status` is created after the config has been loaded. So I created a `/etc/systemd/system/cloud-final.service.d/override.conf` file with the following contents:
```systemd
[Unit]
After=vyos.target

[Service]
ExecStartPre=bash -c 'until test -f /tmp/vyos-config-status; do sleep 1; done'
```

With this config I am able to supply user data like this:
```shell
#!/usr/bin/vbash
source /opt/vyatta/etc/functions/script-template

set system host-name "some-hostname"
commit
save
```

For reference: I used created the following cloud-init configuration for my tests at `/etc/cloud/cloud.cfg.d/90_dpkg.cfg` (I'm running on Hetzner Cloud):
```yaml
datasource_list: [Hetzner, None]

cloud_config_modules:
  - vyos
  - write_files
  - runcmd

cloud_final_modules:
  - scripts_user
```

**Comment by zdc** [comment]
Hi, @jgroom33!
The issue was started a long time ago, but... You can finally test the feature, if it still required: https://github.com/vyos/vyos-cloud-init/pull/27#issue-545607305

**Comment by jgroom33** [comment]
Resolved via #27
